### DISCUSSION TITLE
REPL shows MicroPython v1.13, but platform() shows MicroPython v1.20.0

### DISCUSSION THREAD

**Discussion by inovatorius** [micropython/micropython#12400] [General]
Hello!

Custom build of MicroPython v1.20.0 on ESP8266.

```
MicroPython v1.13-3469-gfcd453529-dirty on 2023-09-08; ESP module (1M) with ESP8266 
Type "help()" for more information.                                                 
>>>
```
However
```
>>> import platform
>>> platform.platform()
'MicroPython-1.20.0-xtensa--with-'
```

**Comment by robert-hh** [comment]
I cannot replicate that. Did you update the tags when pulling the Micropython repository using the --tags option? Like `git pull upstream master --tags` (assuming the MP remote is called upstream).

**Reply by inovatorius** [reply]
I have the MicroPython fork with few commits. I made a fresh clone (via PyCharm) of the fork and built the ESP8266 port. Now I am getting following REPL data:

```
MicroPython d06f70c9b-dirty on 2023-09-10; ESP module (1M) with ESP8266         
Type "help()" for more information.                                             
>>>
```

I did not used the --tags option.

**Reply by inovatorius** [reply]
I pushed tags to fork, updated the project and compiled again:
`MicroPython v1.13-3470-gd06f70c9b on 2023-09-10`

**Reply by andrewleech** [reply]
From your git repo, try running just `git tag` and see what it reports, there should be a long list printed of all the version tags

**Comment by inovatorius** [comment]
> From your git repo, try running just `git tag` and see what it reports, there should be a long list printed of all the version tags

The list is long and repeats:

```
 ESCOD
v1.0
v1.0-rc1
v1.0.1
v1.1
v1.1.1
v1.10
v1.11
v1.12
v1.13
v1.19.1-espnow
v1.2
v1.20.0
 ESCOD
v1.0
v1.0-rc1
v1.0.1
v1.1
v1.1.1
v1.10
v1.11
v1.12
v1.13
v1.19.1-espnow
v1.2
v1.20.0
 ESCOC
```

**Comment by inovatorius** [comment]
Anyway, I made a new MicroPython fork, cloned it, then built the ESP8266:
`MicroPython 3637252b7 on 2023-09-11; ESP module (1M) with ESP8266`

I have no idea how to solve this.

**Reply by robert-hh** [reply]
Run in your copy of micropython directory  the command
git pull --tags
That should get all the tags, displayed with
git tag

**Reply by inovatorius** [reply]
Doesn't solve. I do not remember messing with the tags to compile MicroPython port. 
```
git pull --tags
Already up to date.
git tag

```
`git tag` doesn't show anything, as my fork doesn't have the tags.

Some time ago I had the same issue:
![image](https://github.com/micropython/micropython/assets/122360472/1a7bbace-8790-4540-9a05-467e2438e30e)

I keep making the same mistake, but I have no idea, why I end up with this problems. What GitHub tags have to do with building the port?

**Reply by robert-hh** [reply]
It's that the tags are not updated. Maybe the fork did not copy the tags. Try to add the original Micropython directory as remote with:
`git remote add upstream https://github.com/micropython/micropython.git`
and then run:
`git pull upstream --tags`

**Reply by inovatorius** [reply]
What a headache: 
`MicroPython v1.20.0-450-g3637252b7-dirty on 2023-09-11; ESP module (1M) with ESP8266` 

Thanks @robert-hh and @andrewleech!
